src/Event/Event.cpp

```diff
@@ -49,4 +49,41 @@ namespace LockFreeDispatch {
     {
         dispatched = dispatchStatus;
     }
+
+    Event::Event()
+    {
+        eventID = -1;
+        startTime = Time();
+        durationSeconds = 0;
+        dispatched = false;
+        vehicleRequirementId = -1;
+    }
```

Comment:
```suggestion
    Event::Event()
      : eventID( -1 ), startTime( Time() ), durationSeconds( 0 ), dispatched( false ), vehicleRequirementsId( -1 ) {}
```
Well, I thought that using initialiser lists in constructors instead of the constructor body led to better exception safety, because this object doesn't construct at all if any of the initialiser list elements fail, but I can't find a source for that.

Best I've got is: https://stackoverflow.com/a/926795/2769271